grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -474,11 +476,14 @@ class UserService {
     }
 
     def getDashboardConfig(User user, String id) {
-        def fullConfig = grailsApplication.config.openboxes.dashboardConfig
-        def mainDashboardId = grailsApplication.config.openboxes.dashboardConfig.mainDashboardId
+        Gson gson = new Gson()
+        def fullConfig = gson.fromJson(
+                gson.toJson(Holders.config.openboxes.dashboardConfig),
+                Object
+        )
+        def mainDashboardId = Holders.config.openboxes.dashboardConfig.mainDashboardId
         def resultConfig = [
-                // Clone to not overwrite the main config in the line 422
-                dashboard: fullConfig.dashboards[id ?: mainDashboardId].clone(),
```

Comment:
Why did this stop working? What's the change here between Grails 1?

Reply:
I was trying to find out why it stopped working, but I didn't find anything.
Config properties can be overridden in both cases - in Grails 1 and Grails 3.
In Grails 1, it is also a shallow copy, so honestly, it shouldn't work either.
One difference is just in the implementation of configs - in Grails 1 it is ConfigObject, and in Grails 3 it is NavigableMap